feat: Options to configure file-loader name property

TroAlexis opened the pull request:
It was impossible to configure file-loader with module options. Furthermore, the default configuration was not the one of nuxtjs's. 

Added default configuration for file-loader which matches nuxtjs's destination for images and made possible to pass options to the module. Added generate and start scripts to package.json to test out webpack context.

Module options may either be a function taking context as the argument in case you want to configure different names for files in development and production environment, or just an regular object.

TroAlexis committed:
Added options to module to allow changing svg destination emitted with file-loader. Provided default config with default nuxt options for name option in file-loader.

TroAlexis committed:
feat: Added options to module to allow changing svg destination emitted with file-loader. Provided default config with default nuxt options for name option in file-loader.

TroAlexis committed:
Merge remote-tracking branch 'origin/master'

manuelodelain commented:
Hi @TroAlexis ! 

Thanks for the PR. I would be glad to merge it, could you fix the conflicts please? 

About the default config, I'm not sure but maybe we should add `[path]` to the dev path as Nuxt does?
https://github.com/nuxt/nuxt.js/blob/3ab12f9f104903b2c80813204b114f4103f1cc9b/packages/config/src/config/build.js#L20

Commit:
Merge remote-tracking branch 'original/master'

# Conflicts:
#	lib/module.js
#	package-lock.json

Commit:
fix: rename file-loader options, adapt to new setup function signature

Commit:
fix(nuxt-config): replace require with import as module is exported as es6 module

Commit:
refactor(module): reformat with prettier

Commit:
feat(README): update README configuration section with fileLoader options

TroAlexis commented:
Thanks for your time @manuelodelain  !

Conflicts resolved and you are absolutely right about Nuxt default file-loader name in dev, got that in [04c6c21](https://github.com/nuxt-community/svg-module/pull/75/commits/04c6c216473755f4bf51adb37c8f8317ba69fb94)
Updated the README.md as well to reflect the changes made to configuration.

I also couldn't get to work 'nuxt dev example' to work as the module.js now exports the svgModule function as the default with es6 modules, so I changed nuxt.config.js to import it accordingly if that's all right.
